### PR TITLE
fix(field): making description show up

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Field description wasn't showing up for the internal Field component. This fixes that. 

#### Description of how you validated changes

Added tests

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hein-j — DISMISSED** [review]

File: packages/react/src/primitives/Field/Field.tsx [hunk 1]
```diff
@@ -50,11 +50,12 @@ const FieldPrimitive: Primitive<FieldPrimitiveProps, typeof Flex> = (
       {...rest}
     >
       {label ? <Label visuallyHidden={labelHidden}>{label}</Label> : null}
-      {descriptiveText ? (
```
Comment (reesscot) [inline packages/react/src/primitives/Field/Field.tsx:53]: Wait, why remove this logic? Shouldn't descriptiveText be provided to show the descriptive text?

Comment (zchenwei) [reply packages/react/src/primitives/Field/Field.tsx:53]: +1

Comment (dbanksdesign) [reply packages/react/src/primitives/Field/Field.tsx:53]: That logic is IN `<FieldDescription>`: https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/Field/FieldDescription.tsx#L15

Comment (reesscot) [reply packages/react/src/primitives/Field/Field.tsx:53]: but why did that cause a bug?

Comment (dbanksdesign) [reply packages/react/src/primitives/Field/Field.tsx:53]: This didn't cause the bug, the bug was I was passing it as a child instead of prop

**Review by reesscot — APPROVED** [review]

**Review by hein-j — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]
Thank you for adding tests!

**Pull request merged**